plugin/plugin.go

```diff
@@ -98,90 +96,90 @@ func (p *CLIPlugin) GetMetadata(ctx context.Context, req *proto.GetMetadataReque
 	if metadata.Name != p.name {
 		return nil, fmt.Errorf("executable name must be %q instead of %q", binName(metadata.Name), filepath.Base(p.path))
 	}
-	logger.Debugf("Plugin get-plugin-metadata response: %+v", metadata)
 	return &metadata, nil
 }
 
 // DescribeKey returns the KeySpec of a key.
 //
 // if ContractVersion is not set, it will be set by the function.
 func (p *CLIPlugin) DescribeKey(ctx context.Context, req *proto.DescribeKeyRequest) (*proto.DescribeKeyResponse, error) {
-	logger := log.GetLogger(ctx)
-	logger.Debugf("Plugin describe-key request: %+v", req)
-	var resp proto.DescribeKeyResponse
 	if req.ContractVersion == "" {
 		req.ContractVersion = proto.ContractVersion
 	}
+
+	var resp proto.DescribeKeyResponse
 	err := run(ctx, p.name, p.path, req, &resp)
-	logger.Debugf("Plugin describe-key response: %+v", resp)
 	return &resp, err
```

Comment:
Should we do
```go
var resp proto.DescribeKeyResponse
if err := run(...); err != nil {
    return nil, err
}
return &resp, nil
```
and for all other commands?

Reply:
IMO current format is more compact.

Reply:
But not in a good code style and not memory efficient.